### PR TITLE
Remove traps from authenticated_anchor_operation

### PR DESCRIPTION
**Pull request by frederikrothenberger**
This removes the traps from the authenticated_anchor_operation helper, which allows to add proper errors to `identity_metadata_replace`.

I will do a follow-up PR to move the authentication helpers and error mappers out of the `main.rs` file to keep it focused on the endpoints.

### CODE REVIEW

**Comment by frederikrothenberger** [comment]
> Can you briefly explain why the API breakage is fine?

The API breakage is fine because it is in the API v2 only, which is labeled as experimental.

File: src/internet_identity/src/main.rs [hunk 1]
```diff
@@ -492,6 +501,31 @@ fn authenticate_and_record_activity(anchor_number: AnchorNumber) -> Option<IIDom
...
+    StorageError(IdentityNumber, StorageError),
+}
+
+impl From<IdentityUpdateError> for String {
+    fn from(err: IdentityUpdateError) -> Self {
+        match err {
+            IdentityUpdateError::Unauthorized => {
+                // This error message is used by the legacy API and should not be changed, even though
+                // it is confusing authentication with authorization.
+                format!("{} could not be authenticated.", caller())
```
Comment (przydatek) [inline src/internet_identity/src/main.rs:516]: nit: calling `caller()` in a conversion-to-string function seems weird.  Maybe we should have `IdenityUpdateError::Unauthorized(Principal)` instead? That would also provide more info for debugging/tests.

File: src/internet_identity/src/main.rs [hunk 2]
```diff
@@ -728,20 +765,50 @@ mod v2_api {
...
+            let storage_err = match value {
+                IdentityUpdateError::Unauthorized => {
+                    return IdentityMetadataReplaceError::Unauthorized
+                }
+                IdentityUpdateError::StorageError(_, storage_err) => storage_err,
+            };
+
+            match storage_err {
+                StorageError::EntrySizeLimitExceeded(_) => {
+                    IdentityMetadataReplaceError::StorageSpaceExceeded
```
Comment (przydatek) [inline src/internet_identity/src/main.rs:779]: nit: also here, maybe we should keep the extra information in `StorageSpaceExceeded` (effective size, limit) available for debugging?

**Review by przydatek — APPROVED** [review]
LGTM, modulo two questions.
Thanks!

**Comment by frederikrothenberger** [comment]
@przydatek: I added the context to the errors. Could you please take another look?

I will then also do another PR to also adapt the `PrepareIdAliasError::Unauthorized` variant to hold information about the `principal` that is unauthorized.

**Review by przydatek — APPROVED** [review]
LGTM, thanks!

**Pull request merged**